Epsilon.Abstractions/Component/CompetenceProfile.cs

```diff
@@ -12,17 +13,216 @@ IEnumerable<ProfessionalSkillResult> ProfessionalSkillOutcomes
 {
     public void AddToWordDocument(MainDocumentPart mainDocumentPart)
     {
-        // TODO: This is simply an example to show the capability of the component architecture
         var body = new Body();
 
-        body.AppendChild(
-            new Paragraph(
-                new Run(
-                    new Text("Competence profile comes here")
-                    )
-                )
-            );
+        body.Append(new Paragraph(new Run(new Text(""))));
+        body.AppendChild(GetTasks());
+        body.Append(new Paragraph(new Run(new Text(""))));
+        body.AppendChild(GetSkills());
+        body.Append(new Paragraph(new Run(new Text(""))));
+        body.AppendChild(GetLegend());
+        body.Append(new Paragraph(new Run(new Text(""))));
+
+
 
         mainDocumentPart.Document.AppendChild(body);
     }
+
+    private OpenXmlElement GetTasks()
+    {
+        // Create a table, with rows for the outcomes and columns for the assignments.
+        var table = new Table();
+
+        var tasks = ProfessionalTaskOutcomes
+                    .GroupBy(static task => task.ArchitectureLayer)
+                    .Select(static group => new
+                    {
+                        architectureId = group.Key,
+                        activities = group
+                                     .GroupBy(static act => act.Activity)
+                                     .Select(static group => new
+                                     {
+                                         activityId = group.Key,
+                                         count = group.Count(),
+                                         masteryLevel = group.Max(static m => m.MasteryLevel),
+                                     }).OrderBy(static i => i.activityId).ToList(),
+                    }).OrderBy(static i => i.architectureId).ToList();
+        
+        // Set table properties for formatting.
+        table.AppendChild(new TableProperties(
+            new TableWidth {
+                Width = "8", Type = TableWidthUnitValues.Auto, }));
+
+        var headerRow = new TableRow();
+
+        // Empty top-left cell.
+        headerRow.AppendChild(CreateTableCellWithBorders("2500", new Paragraph(new Run(new Text("")))));
```

Comment:
Could you specify what these values are "`2500`" (maybe make a constant)? Same here with the paragraph stuff.

Reply:
idk what these values are, still need to figure out what they do